lib/socket/socket.nit

```diff
@@ -57,7 +57,7 @@ class TCPStream
 	# Creates a socket connection to host `host` on port `port`
 	init connect(host: String, port: Int)
 	do
-		_buffer = new FlatBuffer
+		_buffer = new NativeString(1024)
```

Comment:
I aligned the size of the buffer on the size of the buffer of the underlying implementation, so yeah it's magic, but not so much.

Though we could change that in the C socket to take a buffer parameter, it can only help with perfs